### PR TITLE
5.0: Update `django.contrib.sitemaps`, `django.contrib.staticfiles`

### PR DESCRIPTION
**Pull request by sudosubin**
# I have made things!

Update stubs for `django.contrib.sitemaps`, and `django.contrib.staticfiles` for Django 5.0.

- [x] `django.contrib.sitemaps.__init__`
  - [x] `django.contrib.sitemaps.PING_URL` was removed
  - [x] `django.contrib.sitemaps.SitemapNotFound` was removed
  - [x] `django.contrib.sitemaps.ping_google` was removed
- [x] `django.contrib.sitemaps.management.commands.ping_google` was removed
- [x] `django.contrib.sites.requests`
  - [x] `django.contrib.sites.requests.RequestSite.save`'s `foce_insert` argument now accept `Union[bool, Tuple[ModelBase, ...]]`
- [x] `django.contrib.staticfiles.checks`
  - [x] `django.contrib.staticfiles.checks.E005` was added
  - [x] `django.contrib.staticfiles.checks.check_storages` was added
- [x] `django.db.models.base`
  - [x] `django.db.models.base.Model.save`'s `foce_insert` argument now accept `Union[bool, Tuple[ModelBase, ...]]`
  - [x] `django.db.models.base.Model.asave`'s `foce_insert` argument now accept `Union[bool, Tuple[ModelBase, ...]]`
  - [x] `django.db.models.base.Model.save_base`'s `foce_insert` argument now accept `Union[bool, Tuple[ModelBase, ...]]`

Updated `django_tests_settings.py`.

```
STATIC_URL = "static/"
```

Fixed some todos.

- [x] `django.contrib.staticfiles.finders.FileSystemFinder.__init__`
- [x] `django.contrib.staticfiles.management.commands.collectstatic`
- [x] `django.contrib.staticfiles.management.commands.findstatic.Command.handle_label`
- [x] `django.contrib.staticfiles.storage`

Changed migrations allowlist location. (`django.contrib.redirects.migrations.*`, `django.contrib.sessions.migrations.*`, `django.contrib.sites.migrations.*`)

## Related issues

Refs #1493

### CODE REVIEW

File: django-stubs/db/models/base.pyi [hunk 1]
```diff
@@ -71,22 +71,22 @@ class Model(metaclass=ModelBase):
     def get_constraints(self) -> list[tuple[type[Model], Sequence[BaseConstraint]]]: ...
     def save(
         self,
-        force_insert: bool = ...,
+        force_insert: bool | tuple[ModelBase] = ...,
```
Comment (flaeppe) [inline django-stubs/db/models/base.pyi]: By looking at the `Model._validate_force_insert` classmethod here: https://github.com/django/django/blob/476d7c581ac97dae2bc14ee79a1294db4f802e74/django/db/models/base.py#L843-L861 I think it's a tuple containing an arbitrary amount of model instances.

Suspect it was just a typo

```suggestion
        force_insert: bool | tuple[ModelBase, ...] = ...,
```

File: scripts/django_tests_settings.py [hunk 2]
```diff
@@ -9,3 +9,5 @@
     "django.contrib.sessions",
     "django.contrib.sites",
 ]
+
+STATIC_URL = "static/"
```
Comment (flaeppe) [inline scripts/django_tests_settings.py:13]: Is it necessary to set the `STATIC_URL` setting?

File: django-stubs/db/models/base.pyi (hunk 1 above)
Comment (sudosubin) [reply django-stubs/db/models/base.pyi]: Oh, that's right. I forgot to add ellipsis 😂  Thanks, I fixed at 847c2ae065a5239c8393849309d6b0e8e5dcf08b.

File: scripts/django_tests_settings.py (hunk 2 above)
Comment (sudosubin) [reply scripts/django_tests_settings.py:13]: Yes. If there is no `STATIC_URL` setting, the following error occurs during stubtest.

```sh
error: django.contrib.staticfiles.management.commands.collectstatic encountered unexpected error, ImproperlyConfigured: You're using the staticfiles app without having set the required STATIC_URL setting.
Stub: in file /app/django-stubs/contrib/staticfiles/management/commands/collectstatic.pyi:1
N/A
Runtime: in file /usr/local/lib/python3.12/site-packages/django/contrib/staticfiles/management/commands/collectstatic.py
This is most likely the fault of something very dynamic in your library. It's also possible this is a bug in stubtest.
If in doubt, please open an issue at https://github.com/python/mypy

Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/django/core/files/storage/handler.py", line 35, in __getitem__
    return self._storages[alias]
           ~~~~~~~~~~~~~~^^^^^^^
KeyError: 'staticfiles'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/mypy/stubtest.py", line 247, in test_module
    yield from verify(stub, runtime, [module_name])
  File "/usr/local/lib/python3.12/site-packages/mypy/stubtest.py", line 404, in verify_mypyfile
    and _belongs_to_runtime(runtime, m)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/mypy/stubtest.py", line 364, in _belongs_to_runtime
    if isinstance(obj, types.ModuleType):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/django/utils/functional.py", line 280, in __getattribute__
    value = super().__getattribute__(name)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/django/utils/functional.py", line 251, in inner
    self._setup()
  File "/usr/local/lib/python3.12/site-packages/django/contrib/staticfiles/storage.py", line 540, in _setup
    self._wrapped = storages[STATICFILES_STORAGE_ALIAS]
                    ~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/django/core/files/storage/handler.py", line 43, in __getitem__
    storage = self.create_storage(params)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/django/core/files/storage/handler.py", line 55, in create_storage
    return storage_cls(**options)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/django/contrib/staticfiles/storage.py", line 29, in __init__
    check_settings(base_url)
  File "/usr/local/lib/python3.12/site-packages/django/contrib/staticfiles/utils.py", line 49, in check_settings
    raise ImproperlyConfigured(
django.core.exceptions.ImproperlyConfigured: You're using the staticfiles app without having set the required STATIC_URL setting.
```

Comment (flaeppe) [reply scripts/django_tests_settings.py:13]: Aha, interesting. Good find

**Review by flaeppe — APPROVED** [review]
Thanks

**Pull request merged**